staging/src/k8s.io/api/admission/v1alpha1/types.go

```diff
@@ -20,52 +20,72 @@ import (
+type AdmissionReviewMetadata struct {
+	// UID is an identifier for the individual request. It allows us to distinguish instances of requests which are
+	// otherwise identical (parallel requests, requests when earlier requests did not modify etc)
+	UID types.UID `json:"uid,omitempty" protobuf:"bytes,1,opt,name=uid,casttype=k8s.io/apimachinery/pkg/types.UID"`
+	// Kind is the type of object being manipulated.  For example: Pod
+	Kind metav1.GroupVersionKind `json:"kind,omitempty" protobuf:"bytes,2,opt,name=kind"`
 	// Name is the name of the object as presented in the request.  On a CREATE operation, the client may omit name and
 	// rely on the server to generate the name.  If that is the case, this method will return the empty string.
-	// +optional
-	Name string `json:"name,omitempty" protobuf:"bytes,5,opt,name=name"`
+	Name string `json:"name,omitempty" protobuf:"bytes,3,opt,name=name"`
 	// Namespace is the namespace associated with the request (if any).
-	// +optional
-	Namespace string `json:"namespace,omitempty" protobuf:"bytes,6,opt,name=namespace"`
+	Namespace string `json:"namespace,omitempty" protobuf:"bytes,4,opt,name=namespace"`
+	// Operation is the operation being performed
+	Operation Operation `json:"operation,omitempty" protobuf:"bytes,5,opt,name=operation"`
 	// Resource is the name of the resource being requested.  This is not the kind.  For example: pods
-	Resource metav1.GroupVersionResource `json:"resource,omitempty" protobuf:"bytes,7,opt,name=resource"`
+	Resource metav1.GroupVersionResource `json:"resource,omitempty" protobuf:"bytes,6,opt,name=resource"`
 	// SubResource is the name of the subresource being requested.  This is a different resource, scoped to the parent
 	// resource, but it may have a different kind. For instance, /pods has the resource "pods" and the kind "Pod", while
 	// /pods/foo/status has the resource "pods", the sub resource "status", and the kind "Pod" (because status operates on
 	// pods). The binding resource for a pod though may be /pods/foo/binding, which has resource "pods", subresource
 	// "binding", and kind "Binding".
-	// +optional
-	SubResource string `json:"subResource,omitempty" protobuf:"bytes,8,opt,name=subResource"`
+	SubResource string `json:"subResource,omitempty" protobuf:"bytes,7,opt,name=subResource"`
 	// UserInfo is information about the requesting user
-	UserInfo authenticationv1.UserInfo `json:"userInfo,omitempty" protobuf:"bytes,9,opt,name=userInfo"`
+	UserInfo authenticationv1.UserInfo `json:"userInfo,omitempty" protobuf:"bytes,8,opt,name=userInfo"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AdmissionReviewResponse describes an admission response.
+type AdmissionReviewResponse struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Metadata contains metadata about the request including the UID from the Request.
+	Metadata AdmissionReviewMetadata `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
```

Comment:
surprising to have the same metadata again as in the request. Why?

Reply:
Explanation in #55729. But likely we are going with plan b on this one.